google/ggrc-core
# GGRC-4711 Store all ACL propagation data in access_control_roles table

# Dependencies

This PR is `on hold` until the dependencies are merged:

- [x] https://github.com/google/ggrc-core/pull/7495
- [x] https://github.com/google/ggrc-core/pull/7510
- [x] https://github.com/google/ggrc-core/pull/7511
- [ ] https://github.com/google/ggrc-core/pull/7503

# TODO:

- [x] Create ticket for this PR and update the title.
- [ ] N/A Roles for Risk Assessments. - apparently we don't have these migrated to ACL yet.
- [x] Roles for Workflow objects.
- [x] Non-admin roles on all objects

# Issue description

We have to specify the full role propagation rules in Mysql so that we can create efficient queries for ACL propagation.

This PR is not yet fully finished but it is opened because the review should take a long time to verify we did not forget any propagation rules and that we did not add too many rules.

# Steps to test the changes

Read the propagation tree created in the database and verify that we have all needed rules and that no rule is missing.

This is going to be hard since we already have RBAC bugs, so in the investigation check with Liliya, QA team, and the new RBAC/ACL documentation. 

Any discrepancies must be discussed and the correct behavior must be confirmed by at least of the above-mentioned ways of verification.

# Solution description

We created a migration utility that can receive a specially formatted dictionary and that in turn creates or deletes the new proper propagation rows. 

Then we created a dictionary containing all propagation rules for all objects and all object non-editable roles.


# Sanity checklist

- [ ] N/A My changes are covered by tests. - No functionality changes happened and we usually don't have any tests for migrations.

# Migration checklist



# PR Review checklist
